freecad_api/main.py

```
@@ -397,622 +158,136 @@ async def convert_to_2d(file: Optional[UploadFile] = File(None), grid_data: Opti
-            with open(temp_model_script.name, "w") as f:
-                f.write(
-                    f"""
-import json
+        # 一時ファイルにアップロードされたファイルを保存
+        temp_file = tempfile.NamedTemporaryFile(suffix=".fcstd", delete=False)
+        content = await file.read()
+        temp_file.write(content)
+        temp_file.close()
+
+        # 一時ファイルにFreeCADスクリプトを保存
+        temp_script = tempfile.NamedTemporaryFile(suffix=".py", delete=False)
```

Ruff (0.8.2)

Use a context manager for opening files

(SIM115)